test_unstructured_ingest/test-ingest-elasticsearch.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd "$SCRIPT_DIR"/.. || exit 1
+
+# Create the elasticsearch cluster and get the container id
+output=$(docker run -d --rm -p 9200:9200 -p 9300:9300 -e "xpack.security.enabled=false" -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:8.7.0)
+container_id=$(echo "$output" | cut -c 1-12)
+
+url="http://localhost:9200/_cluster/health"
+status_code=0
+retry_count=0
+max_retries=6
+
+# Check the cluster status repeatedly until it becomes live or maximum retries are reached
+while [ "$status_code" -ne 200 ] && [ "$retry_count" -lt "$max_retries" ]; do
+
+  # Send a GET request to the cluster health API
+  response=$(curl -s -o /dev/null -w "%{http_code}" "$url")
+  status_code="$response"
+
+  # Process the files only when the ES cluster is live
+  if [ "$status_code" -eq 200 ]; then
+    echo "Cluster is live."
+
+    python examples/ingest/elasticsearch/elasticsearch_cluster.py
```

Comment:
can we move all of this code to launch the es cluster into its own shell script? 
I think it'll be cleaner, but extra benefit that then we don't repeat this in the example ingest.sh as well

Reply:
EDIT: actually, include all the way up to line 6 (so, all logic needed to launch it)

Reply:
EDIT: and code lower down where we exit on retries and stop the docker container. basically everything to encapsulate launching the test es cluster

Reply:
I isolated all code in another script, except the line shutting the container down

Had to trap the shut-down in this script since we only want it shut down after this script exits

If we'd like to isolate that line as well, I can create another script solely for shutting the container down

Reply:
nope, I think that makes perfect sense!